src/components/Auth.tsx

```diff
@@ -1,47 +1,115 @@
-var firebaseConfig = {
+const config = {
   apiKey: "AIzaSyCvl1CTEcEWYM1681gUWSaawnHAV-PEgWo",
   authDomain: "my-recipes-da233.firebaseapp.com",
   databaseURL: "https://my-recipes-da233.firebaseio.com",
   projectId: "my-recipes-da233",
-  storageBucket: "my-recipes-da233.appspot.com",
-  messagingSenderId: "1062905305210",
   appId: "1:1062905305210:web:e81f2b5d293bedc736c3c3",
   measurementId: "G-BJ3YHDCKZV",
 };
-// Initialize Firebase
-firebase.initializeApp(firebaseConfig);
-firebase.analytics();
-
-var uiConfig = {
-  signInSuccessUrl: "https://alex-aranda.com",
-  signInOptions: [
-    // Leave the lines as is for the providers you want to offer your users.
-    firebase.auth.EmailAuthProvider.PROVIDER_ID,
-  ],
-  // tosUrl and privacyPolicyUrl accept either url string or a callback
-  // function.
-  // Terms of service url/callback.
-  tosUrl: "<your-tos-url>",
-  // Privacy policy url/callback.
-  privacyPolicyUrl: function () {
-    window.location.assign("<your-privacy-policy-url>");
-  },
-};
+firebase.initializeApp(config);
+
+interface authProps {
+  displayName: string;
+  isSignedIn: boolean;
+  signIn: (d: any, e: any, u: any) => void;
+  signOut: () => void;
+}
+
+class Auth extends Component<authProps> {
+  constructor(props: authProps) {
+    super(props);
+    this.state = { displayName: "", email: "", uid: "", isSignedIn: false };
+  }
+  uiConfig = {
+    callbacks: {
+      signInSuccessWithAuthResult: () => {
+        return false;
+      },
+    },
+    signInOptions: [firebase.auth.EmailAuthProvider.PROVIDER_ID],
+  };
 
-// Initialize the FirebaseUI Widget using Firebase.
-var ui = new firebaseui.auth.AuthUI(firebase.auth());
-// The start method will wait until the DOM is loaded.
-ui.start("#firebaseui-auth-container", uiConfig);
+  componentDidMount() {
+    firebase.auth().onAuthStateChanged((user) => {
+      if (user) {
+        this.props.signIn(user.displayName, user.email, user.uid);
+        this.setState({
+          displayName: user.displayName,
+          email: user.email,
+          uid: user.uid,
+          isSignedIn: true,
+        });
+      } else {
+        console.log("no user");
+      }
+    });
+  }
 
-function Auth() {
-  return <div id="firebaseui-auth-container"></div>;
+  render() {
+    if (!this.props.isSignedIn) {
+      return (
+        <div>
+          <h1>My Recipes</h1>
+          <p>Please sign-in:</p>
+          <StyledFirebaseAuth
+            uiConfig={this.uiConfig}
+            firebaseAuth={firebase.auth()}
+          />
+        </div>
+      );
+    }
+    return (
+      <div>
+        <h1>My App</h1>
+        <p>Welcome {this.props.displayName}! You are now signed-in!</p>
+        <div
+          onClick={() => {
+            firebase.auth().signOut();
+            this.props.signOut();
+          }}
+        >
+          Sign-out
+        </div>
+      </div>
+    );
+  }
 }
 
-export default Auth;
+interface mapState {
+  countReducer: {

```

Comment:
It would probably make sense to define and export your reducer type in the reducer's file.  Then in your rootReducer, you can export an interface for `RootState` that uses the reducer types.